information/specification.md

```diff
@@ -0,0 +1,826 @@
+set](#remaining-utxo-set), but _not both_.
+
+  * If a UTxO entry _is_ selected by the coin selection algorithm, it is
+    included in the [coin selection](#coin-selection) inputs.
+
+  * If a UTxO entry is _not_ selected by the coin selection algorithm, it is
+    included in the [remaining UTxO set](#remaining-utxo-set).
+
+The following laws hold:
+
+  * UTxO<sub>_selected_</sub> ⊆ UTxO<sub>_initial_</sub>
+  * UTxO<sub>_remaining_</sub> ⊂ UTxO<sub>_initial_</sub>
+  * UTxO<sub>_remaining_</sub> ∩ UTxO<sub>_selected_</sub> = ∅
+  * UTxO<sub>_remaining_</sub> ⋃ UTxO<sub>_selected_</sub> =
+    UTxO<sub>_initial_</sub>
+
+Where:
+
+  * **UTxO<sub>_initial_</sub>**
+
+    is the [initial UTxO set](#initial-utxo-set).
+
+  * **UTxO<sub>_remaining_</sub>**
+
+    is the [remaining UTxO set](#remaining-utxo-set).
+
+  * **UTxO<sub>_selected_</sub>**
+
+    is the value of the _inputs_ field of the [coin selection](#coin-selection)
+    result.
+
+### Conservation of Outputs
+
+This property states that the [requested output list](#requested-output-list)
+is _conserved_ in the [coin selection](#coin-selection) result.
+
+In particular, the _outputs_ field of the [coin selection](#coin-selection)
+result should be _equal to_ the [requested output list](#requested-output-list).
+
+## Failure Modes
+
+There are a number of ways in which a coin selection algorithm can fail:
+
+  * #### UTxO Balance Insufficient
+
+    This failure occurs when the total value of the entries within the [initial
+    UTxO set](#initial-utxo-set) (the amount of money _available_) is _less
+    than_ the the total value of all entries in the [requested output
+    list](#requested-output-list) (the amount of money _required_).
+
+  * #### UTxO Not Fragmented Enough
+
+    This failure occurs when the _number_ of entries in the [initial UTxO
+    set](#initial-utxo-set) is _smaller than_ the number of entries in the
+    [requested output list](#requested-output-list), for algorithms that impose
+    the restriction that a single UTxO entry can only be used to pay for _at
+    most one_ output.
+
+  * #### UTxO Fully Depleted
+
+    This failure occurs if the algorithm depletes all entries from the [initial
+    UTxO set](#initial-utxo-set) _before_ it is able to pay for all outputs in
+    the [requested output list](#requested-output-list).
+
+    This can happen _even if_ the total value of entries within the [initial
+    UTxO set](#initial-utxo-set) is _greater than_ the total value of all
+    entries in the [requested output list](#requested-output-list), due to
+    various restrictions that coin selection algorithms impose on themselves
+    when selecting UTxO entries.
+
+  * #### Maximum Input Count Exceeded
+
+    This failure occurs if the _number_ of UTxO entries needed to pay for the
+    outputs in the [requested output list](#requested-output-list) exceeds the
+    upper limit specified by the [maximum input count](#maximum-input-count)
+    parameter.
```

Comment:
The maximum input count parameter is more of a symptom IMO. This doesn't really tell you why it is wrong to select too many entries (although mentioned in previous sections). What about:

"This failure occurs when another input must be selected by the algorithm to continue but doing so will increase the size of the resulting transaction beyond an acceptable limit."

Reply:
> This failure occurs when another input must be selected by the algorithm to continue but doing so will increase the size of the resulting transaction beyond an acceptable limit.

Thanks. I've incorporated this suggestion into the following change: e31e212078ded1f1b59804f245d4d8c6683b07b5